jax/abstract_arrays.py

```diff
@@ -208,3 +226,19 @@ def raise_to_shaped(aval):
     raise TypeError(type(aval))
 
 core.literalable_types.update(array_types)
```

Comment:
Since `python_scalar_dtypes` are no longer included in `array_types`, and since there's no other code updating `core.literalable_types`, IIUC after this change Python scalar types might not be able to be turned into literals anymore. Is that right? And if so, can we update `core.literalable_types` with `python_scalar_types` wherever appropriate?

I realize it's problematic that we don't have a test for what gets turned into an in-line literal in jaxprs, but I think we can check `self.assertLen(make_jaxpr(lambda x: x + 2).constvars, 2)`.

Reply:
I made scalars literalable once again. The test verifies that `constvars` is 0, not 2. I guess that's what you meant?

Reply:
Yes, typo!